specs/AppLifecycle/Activation/AppLifecycle Activation.md

```diff
@@ -0,0 +1,434 @@
+        ::OutputDebugString(sStream.str().c_str());
+    }
+}
+```
+
+<br>
+
+## API Details
+
+### Top-level types
+
+This spec proposes one new enum and 6 new classes:
+
+- An ExtendedActivationKind enum, based off
+  Windows.ApplicationModel.Activation.ActivationKind.
+- An ActivationArguments class that brings forward the core behavior of the platform's IActivatedEventArgs interface.
+- An AppInstance class to encapsulate all API functionality for an instance of an app.
+- An XXXActivatedEventArgs class for each activation kind, modeled directly on existing XXXActivatedEventArgs classes in the Windows.ApplicationModel.Activation namespace.
+
+| API                                                          | Description                                                  |
+| ------------------------------------------------------------ | ------------------------------------------------------------ |
+| enum class ExtendedActivationKind<br/>{<br/>    /\* all existing platform ActivationKind values, \*/<br/>    ExtensionBase = 5000,<br/>    /\* new Reunion ExtendedActivationKind values\*/<br/>}; | This enum is based off the platform [ActivationKind](https://docs.microsoft.com/en-us/uwp/api/Windows.ApplicationModel.Activation.ActivationKind). All platform ActivationKind values are cloned to this class. Then, we set a high-value **ExtensionBase** value which is well above the existing highest ActivationKind value (1026), to allow for new values to be added to the platform. Going forward, any new platform values will also be added to the new enum, plus any new values that are only defined in the new enum. |
+| class ActivationArguments<br/>{<br/>public:<br/>    ExtendedActivationKind Kind ;<br/>    IInspectable Data;<br/>}; | The platform defines [IActivatedEventArgs](https://docs.microsoft.com/en-us/uwp/api/Windows.ApplicationModel.Activation.IActivatedEventArgs), which exposes a Kind property. Apps can get an IActivatedEventArgs reference, examine the Kind and then cast the reference to the specific concrete class type that implements IActivatedEventArgs. This enables generic functions such as GetActivatedEventArgs and the Activated event handler, which provide the app with an IActivatedEventArgs that the app can then convert to a specific concrete type. However, IActivatedEventArgs also defines other properties that are only useful in a UWP context. In addition, we need a mechanism that supports both platform ActivationKinds and also Reunion-only ExtendedActivationKinds and their corresponding concrete class types. <br />The new ActivationArguments type serves this purpose: the app can examine the Kind, and then cast the Data property to the specific concrete type (which can be either a platform IActivatedEventArgs concrete class, or a Reunion class).<br />**Kind** is the enum value that indicates the kind of activation this object represents.<br />**Data** can be cast to a specific concrete class corresponding to Kind. |
```

Comment:
So if we need to add a new PushActivatedKind which isn't part of Windows.Foundation today, we should be able to technically throw this into the IInspectable ref over here correct and add a new ActivationKindExtension?
@danielayala94  FYI

Reply:
@hulumane Take a look at my instancing topic branch for a better idea of where the implementation is going at least.  It should match this, but has some examples.